pkg/eks/tasks.go

```diff
@@ -431,3 +431,23 @@ func (c *ClusterProvider) appendCreateTasksForIAMServiceAccounts(cfg *api.Cluste
 		spec:            cfg,
 	})
 }
+
+// clusterSupportsDevicePlugin returns true if any of the node/managed node groups
+// uses an AMI family which requires the device plugin to be installed as
+// a daemonset
+func clusterSupportsDevicePlugin(cfg *api.ClusterConfig) bool {
+	supported := false
+	for _, ng := range cfg.NodeGroups {
+		if supported {
+			break
+		}
+		supported = ng.NodeGroupBase.GetAMIFamily() != api.NodeImageFamilyBottlerocket
+	}
+	for _, mng := range cfg.ManagedNodeGroups {
+		if supported {
+			break
+		}
+		supported = mng.NodeGroupBase.GetAMIFamily() != api.NodeImageFamilyBottlerocket
+	}
+	return supported
```

Comment:
Only AL2 GPU AMIs require the device plugin, so the condition can be relaxed. Additionally, `NodeGroupBase` is an embedded field, so `GetAMIFamily` (or `AMIFamily`) can be accessed directly. 

```suggestion
	for _, ng := range cfg.NodeGroups {
		if ng.GetAMIFamily() == api.NodeImageFamilyAmazonLinux2 {
			return true
		}
	}
	for _, mng := range cfg.ManagedNodeGroups {
		if mng.GetAMIFamily() == api.NodeImageFamilyAmazonLinux2 {
			return true
		}
	}
	return false
```